.github/workflow/pylint.yml

```diff
@@ -0,0 +1,23 @@
+name: Pylint
```

Comment:
once you test if these two work, it'll make logical sense to combine them into a "lint" step.  So run all linters together; you can run one job and then another rather than two separate pipelines.

Reply:
So the lint "step" would be another directory inside "workflows"? Does GitHub automatically search through the .github directory for any relevant files each time you "do" something like a pull request, edit, etc?

Reply:
yes, github automatically searchines `.github/workflows` for yml files and tries to parse them into it's workflows.  Each time you make a change to these, it picks up the new changes and tries to execute them.  

you can have one `lint.yml` file that has multiple steps under the `jobs:` chunk.  [docs ref](https://docs.github.com/en/actions/using-jobs/using-jobs-in-a-workflow#example-creating-jobs).  I would do the pylint job first, then the black.  Though since they're both python formatters, you could actually just run one than the other.  

```yaml
name: Lint

on: [push, pull-request]

jobs:
  build:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        python-version: ["3.8", "3.9", "3.10"]
    steps:
    - uses: actions/checkout@v3
    - name: Set up Python ${{ matrix.python-version }}
      uses: actions/setup-python@v3
      with:
        python-version: ${{ matrix.python-version }}
    - name: Install dependencies
      run: |
        python -m pip install --upgrade pip
        pip install pylint black
    - name: Analysing the code with pylint
      run: |
        pylint $(git ls-files '*.py')
    - name: Analysing the code with black
    - run: black . 
```